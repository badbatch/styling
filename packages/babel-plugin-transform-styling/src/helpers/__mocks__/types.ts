import { StylingNamedExports } from "../../types";

export interface MockBuildTransformedFile {
  _setFile: (file: string) => void;
  default: (namedExports: StylingNamedExports) => string;
}

export interface MockEvalStylingFile {
  _setFile: (file: StylingNamedExports) => void;
  default: (filename: string) => StylingNamedExports;
}
