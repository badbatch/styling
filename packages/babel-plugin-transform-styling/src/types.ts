import { NodePath, Visitor } from "@babel/traverse";
import { StringObject } from "@styling/types";

export type ExportsArgsMap = Map<string, { type: string; value: string }>;

export interface ExportsArgsResult {
  identifiers: string[];
  map: ExportsArgsMap;
}

export interface PluginState {
  cwd: string;
  file: {
    path: NodePath;
  };
  filename: string;
}

export interface PluginResult {
  visitor: Visitor<PluginState>;
}

export interface StylingNamedExport {
  propsToClassNamesMap: StringObject;
}

export interface StylingNamedExports {
  [key: string]: StylingNamedExport;
}
