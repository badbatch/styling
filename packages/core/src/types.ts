import {
  ComponentType,
  ForwardRefExoticComponent,
  PropsWithChildren,
  ReactHTML,
  ReactNode,
  ReactSVG,
  RefForwardingComponent,
} from "react";

export type ForwardRefComponent<P extends object> = ForwardRefExoticComponent<P> & {
  readonly $$typeof: symbol;
  render: RefForwardingComponent<any, PropsWithChildren<{}>>; // tslint:disable-line no-any
};

export interface ForwardedProps {
  as?: ComponentType | keyof ReactHTML | keyof ReactSVG;
  children?: ReactNode;
  className?: string;
  [key: string]: any; // tslint:disable-line no-any
}

export interface ReturnedElementProps {
  children?: ReactNode;
  className?: string;
  ref?: React.Ref<any>; // tslint:disable-line no-any
  stylingClassNames?: string;
  [key: string]: any; // tslint:disable-line no-any
}

export type StylingComponent<P extends object> = ForwardRefComponent<P> & {
  component?: StylingComponentOrTagType<P>;
  isStyling?: true;
};

export type StylingComponentOrTagType<P extends object> =
  | ComponentType
  | ForwardRefComponent<P>
  | StylingComponent<P>
  | keyof ReactHTML
  | keyof ReactSVG;

export type StylingComponentType = ComponentType | ForwardRefComponent<{}> | StylingComponent<{}>;
