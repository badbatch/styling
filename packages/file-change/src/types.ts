import { PlainObject } from "@styling/types";

export interface CheckFileAndImportsOptions {
  lastCheckedFiles: PlainObject;
  parentLastCheckedTimestamp?: number;
}
