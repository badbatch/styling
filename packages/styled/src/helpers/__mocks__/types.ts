export interface MockBuildBaseSelector {
  _setSelector: (selector: string) => void;
  default: (componentName: string, prefix?: string) => string;
}
