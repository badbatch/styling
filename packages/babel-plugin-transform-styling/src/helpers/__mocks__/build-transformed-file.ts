import { MockBuildTransformedFile, StylingNamedExports } from "../../types";

const mockModule = jest.genMockFromModule("../build-transformed-file") as MockBuildTransformedFile;

let mockFile: string;

function _setFile(file: string) {
  mockFile = file;
}

function buildTransformedFile(namedExports: StylingNamedExports) {
  return mockFile;
}

mockModule._setFile = _setFile;
mockModule.default = buildTransformedFile;

module.exports = mockModule;
