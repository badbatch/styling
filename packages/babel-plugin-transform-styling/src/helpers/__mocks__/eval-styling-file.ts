import { MockEvalStylingFile, StylingExports } from "../../types";

const mockModule = jest.genMockFromModule("../eval-styling-file") as MockEvalStylingFile;

let mockFile: StylingExports;
let originalFile: string;

function _getOriginal() {
  return originalFile;
}

function _setMock(file: StylingExports) {
  mockFile = file;
}

function evalStylingFile(file: string) {
  originalFile = file;
  return mockFile;
}

mockModule._getOriginal = _getOriginal;
mockModule._setMock = _setMock;
mockModule.default = evalStylingFile;

module.exports = mockModule;
