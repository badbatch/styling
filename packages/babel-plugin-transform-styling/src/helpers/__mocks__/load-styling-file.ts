import { StylingNamedExports } from "../../types";
import { MockLoadStylingFile } from "./types";

const mockModule = jest.genMockFromModule("../build-transformed-file") as MockLoadStylingFile;

let mockFile: StylingNamedExports;

function _setFile(file: StylingNamedExports) {
  mockFile = file;
}

function loadStylingFile(filename: string) {
  return mockFile;
}

mockModule._setFile = _setFile;
mockModule.default = loadStylingFile;

module.exports = mockModule;
