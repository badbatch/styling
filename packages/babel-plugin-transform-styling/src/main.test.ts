import { transform } from "@babel/core";
import { MockEvalStylingFile } from "./helpers/__mocks__/types";
import transformStylingFiles from "./main";

jest.mock("./helpers/eval-styling-file");
const evalStylingFile = require("./helpers/eval-styling-file") as MockEvalStylingFile;

describe("transformStylingFiles >>", () => {
  const sourceFile = `
    import { styled } from "@styling/core";
    import BlackText from "./text";
    import { BlueSubtext, RedSubtext } from "./subtext";

    export const Container = styled(
      "div",
      ["checked", "disabled"],
    )\`
      display: block;
    \`;

    export const Radio = styled(
      "input",
      ["checked", "disabled"],
    )\`
      display: block;
    \`;

    export const Text = styled(
      BlackText,
      ["checked", "disabled"],
    )\`
      display: block;
    \`;

    export const Subtext = styled(
      BlueSubtext,
      ["checked", "disabled"],
    )\`
      display: block;
    \`;
  `;

  beforeEach(() => {
    evalStylingFile._setFile({
      Container: {
        propsToClassNamesMap: {
          checked: "test-component__container--checked",
          "checked::disabled": "test-component__container--checked--disabled",
          disabled: "test-component__container--disabled",
        },
      },
      Radio: {
        propsToClassNamesMap: {
          checked: "test-component__radio--checked",
          "checked::disabled": "test-component__radio--checked--disabled",
          disabled: "test-component__radio--disabled",
        },
      },
      Subtext: {
        propsToClassNamesMap: {
          checked: "test-component__subtext--checked",
          "checked::disabled": "test-component__subtext--checked--disabled",
          disabled: "test-component__subtext--disabled",
        },
      },
      Text: {
        propsToClassNamesMap: {
          checked: "test-component__text--checked",
          "checked::disabled": "test-component__text--checked--disabled",
          disabled: "test-component__text--disabled",
        },
      },
    });
  });

  it("SHOULD generate the correct output for the correct files", () => {
    const result = transform(sourceFile, { filename: "index.styling.ts", plugins: [transformStylingFiles] });

    if (result) {
      expect(result.code).toMatchSnapshot();
    }
  });
});
