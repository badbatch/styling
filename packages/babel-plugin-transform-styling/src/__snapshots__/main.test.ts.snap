// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformStylingFiles >> SHOULD generate the correct output for the correct files 1`] = `
"import styling from \\"@styling/core\\";
import BlackText from \\"path/to/app-root/src/component/text\\";
import { BlueSubtext } from \\"path/to/app-root/src/component/subtext\\";
export const Container = styling(\\"div\\", [\\"checked\\", \\"disabled\\"], [\\"checked\\", \\"disabled\\"], {
  \\"checked\\": \\"test-component__container--checked\\",
  \\"checked--disabled\\": \\"test-component__container--checked--disabled\\",
  \\"disabled\\": \\"test-component__container--disabled\\"
});
export const Radio = styling(\\"input\\", [\\"checked\\", \\"disabled\\"], [\\"checked\\", \\"disabled\\"], {
  \\"checked\\": \\"test-component__radio--checked\\",
  \\"checked--disabled\\": \\"test-component__radio--checked--disabled\\",
  \\"disabled\\": \\"test-component__radio--disabled\\"
});
export const Subtext = styling(BlueSubtext, [\\"checked\\", \\"disabled\\"], [\\"checked\\", \\"disabled\\"], {
  \\"checked\\": \\"test-component__subtext--checked\\",
  \\"checked--disabled\\": \\"test-component__subtext--checked--disabled\\",
  \\"disabled\\": \\"test-component__subtext--disabled\\"
});
export const Text = styling(BlackText, [\\"checked\\", \\"disabled\\"], [\\"checked\\", \\"disabled\\"], {
  \\"checked\\": \\"test-component__text--checked\\",
  \\"checked--disabled\\": \\"test-component__text--checked--disabled\\",
  \\"disabled\\": \\"test-component__text--disabled\\"
});"
`;

exports[`transformStylingFiles >> SHOULD pass the correct code to evalStylingFile 1`] = `
"let styled;

try {

    const require_0 = require(\\"@styling/core\\");
    styled = require_0.styled
  ;
} catch {}

let BlackText;

try {

    const require_1 = require(\\"/Users/dylanaubrey/Documents/workspaces/styling/src/component/text\\");
    BlackText = require_1.default
  ;
} catch {}

let BlueSubtext;

try {

    const require_2 = require(\\"/Users/dylanaubrey/Documents/workspaces/styling/src/component/subtext\\");
    BlueSubtext = require_2.BlueSubtext
  ;
} catch {}

let consts;

try {

    consts = require(\\"/Users/dylanaubrey/Documents/workspaces/styling/src/component/constants\\");
  ;
} catch {}

export const foo = \\"bar\\";
export const Container = styled(\\"div\\", [\\"checked\\", \\"disabled\\"], \\"Container\\", \\"/Users/dylanaubrey/Documents/workspaces/styling/src/component/index.styling.ts\\")\`
    display: block;
  \`;
export const Radio = styled(\\"input\\", [\\"checked\\", \\"disabled\\"], \\"Radio\\", \\"/Users/dylanaubrey/Documents/workspaces/styling/src/component/index.styling.ts\\")\`
    display: block;
  \`;
export const Text = styled(BlackText, [\\"checked\\", \\"disabled\\"], \\"Text\\", \\"/Users/dylanaubrey/Documents/workspaces/styling/src/component/index.styling.ts\\")\`
    display: block;
  \`;
export const Subtext = styled(BlueSubtext, [\\"checked\\", \\"disabled\\"], \\"Subtext\\", \\"/Users/dylanaubrey/Documents/workspaces/styling/src/component/index.styling.ts\\")\`
    display: block;
  \`;"
`;
